django/thunderstore/core/utils.py

```diff
@@ -23,3 +32,42 @@ def get_serializer_context(self):
         context = super().get_serializer_context()
         context["community_site"] = self.request.community_site
         return context
+
+
+def capture_exception(exception: Exception) -> None:
+    """Raises exception when running tests or NO_SILENT_FAIL."""
+    testing = "PYTEST_CURRENT_TEST" in os.environ
+    if testing or settings.NO_SILENT_FAIL:
+        raise exception
+    if not testing and settings.SENTRY_DSN:
+        capture_sentry_exception(exception)
+
+
+def _create_ghost_user_username(id_: str) -> str:
+    return f"Ghost User {id_}"
+
+
+def _create_ghost_user_email(id_: str) -> str:
+    # The ID used in ghost user usernames and emails are not the same as their `User.id`
+    # This is because `User` still uses an autoincrement ID
+    return f"{id_}.gu@thunderstore.io"
+
+
+@transaction.atomic
+def delete_user(user: User) -> None:
+    """
+    Delete a User.
+
+    This should be the only place where User.delete() is called.
+    """
+    # Ghost user
+    if user.comments.exists():
+        uuid = generate_ulid_as_uuid()
+        username = _create_ghost_user_username(uuid.hex)
+        email = _create_ghost_user_email(uuid.hex)
+        ghost_user = User.objects.create_user(username, email=email)
+        for comment in user.comments.iterator():
+            comment.author = ghost_user
+            comment.save()
```

Comment:
Might be able to improve performance here. Related: https://github.com/risk-of-thunder/Thunderstore/pull/169#discussion_r546434926

Reply:
`comment.save(update_fields=("author",))` will help at the very least.

If this turns out to be a bigger task than we expected, we should consider making it a celery task and run it as a background job

Reply:
Actually yes, let's turn this into a celery task instead, as that's the best practice going forward